src/ImageRunModal.jsx

```diff
@@ -474,19 +474,46 @@ export class ImageRunModal extends React.Component {
 
         this.setState({ searchFinished: false, searchInProgress: true });
         this.activeConnection = rest.connect(client.getAddress(this.state.isSystem), this.state.isSystem);
+        let searches = [];
+
+        // If there are registries configured search in them, or if a user searches for `docker.io/cockpit` let
+        // podman search in the user specified registry.
+        if (Object.keys(this.props.registries).length !== 0 || value.includes('/')) {
+            searches.push(this.activeConnection.call({
+                method: "GET",
+                path: client.VERSION + "libpod/images/search",
+                body: "",
+                params: {
+                    term: value,
+                }
+            }));
+        } else {
+            searches = searches.concat(utils.fallbackRegistries.map(registry =>
+                this.activeConnection.call({
+                    method: "GET",
+                    path: client.VERSION + "libpod/images/search",
+                    body: "",
+                    params: {
+                        term: registry + "/" + value
+                    }
+                })));
+        }
 
-        const options = {
-            method: "GET",
-            path: client.VERSION + "libpod/images/search",
-            body: "",
-            params: {
-                term: value,
-            },
-        };
-        this.activeConnection.call(options)
+        Promise.allSettled(searches)
                 .then(reply => {
                     if (reply && this._isMounted) {
-                        const imageResults = JSON.parse(reply);
+                        let imageResults = [];
+                        let dialogError = "";
+                        let dialogErrorDetail = "";
+
+                        for (const result of reply) {
+                            if (result.status === "fulfilled") {
+                                imageResults = imageResults.concat(JSON.parse(result.value));
+                            } else {
+                                dialogError = _("Failed to search for new images");
+                                dialogErrorDetail = cockpit.format(_("Failed to search for images: $0"), result.reason);
```

Comment:
It'd be nice to have the registry in the error message. Perhaps it's already part of result.reason. Can be a follow-up for sure.

Reply:
That would be nice, but not included in the reply from podman.

Reply:
No, but we know which registry we put into the search API call.